core/iwasm/interpreter/wasm_loader.c

```diff
@@ -6333,8 +6334,19 @@ create_sections(const uint8 *buf, uint32 size, WASMSection **p_section_list,
             }
 
             section->section_type = section_type;
-            section->section_body = (uint8 *)p;
             section->section_body_size = section_size;
+            if (alloc_mem_for_section_body) {
+                section->section_body = loader_malloc(
+                    section->section_body_size, error_buf, error_buf_size);
```

Comment:
Same as aot, should change destroy_sections in this file.

Reply:
Is it needed? `destroy_sections` is only called by `load`, not by `create_sections`

Reply:
It is to avoid memory leak, `destroy_sections` is called when `create_sections` failed: some section_bodies may have been allocated when a section body is allocated failed.

Reply:
I know why now, in this function, when creating section failed, it doesn't free the section nodes created like AOT, and the below code in function load will call destroy_sections to free them even when `create_sections` failed:
```C
    if (!create_sections(buf, size, false, &section_list, error_buf,
                         error_buf_size)
        || !load_from_sections(module, section_list, true, error_buf,
                               error_buf_size)) {
        destroy_sections(section_list);
        return false;
    }
```
But I think it is not a good coding style, and it will lead to memory leak in the wasm_read_to_sections, which doesn't call destroy_sections when creating sections failed:
```C
    return create_sections(buf, size, true, p_section_list, error_buf,
                           error_buf_size);
```
My suggestion is to do like aot: call `destroy_sections` in the lable `fail` when failing to `create_sections`, and change `return false` in this function to `goto fail`. And also add an argument `free_section_body` like aot.

And change below code to be like:
```C
    if (!create_sections(buf, size, false, &section_list, error_buf,
                         error_buf_size))
        return false;

    if (!load_from_sections(module, section_list, true, error_buf,
                               error_buf_size)) {
        destroy_sections(section_list, false); //free_section_body = 
        return false;
    }

    destroy_sections(section_list, false); //free_section_body = false
    return true;
```